### PR TITLE
fix analytics conversion export

### PR DESCRIPTION
**Pull request by resslinger**
### 1. Why is this change necessary?
Currently, only the entries from the first page are exported from the "Conversion Overview" report type.
This PR fix the csv export of this report type.

### 2. What does this change do, exactly?
The change export all entries.

### 3. Describe each step to reproduce the issue or behaviour.
1.) Go into the backend.
2.) Open the window marketing > evaluations > evaluations.
3.) Select the type "Conversion Overview"
4.) Filter for more then 3 month, for example.
5.) Click on export

=> in the export is only the result of the first page included.

With this fix, all entries are exported as csv now.

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Controllers/Backend/Analytics.php [hunk 1]
```diff
@@ -279,11 +279,17 @@ public function getRatingAction()
             }
         }
 
+        // Sets the correct limit
+        $limit = 25;
+        if (strtolower($this->format) === 'csv') {
```
Comment (shyim) [inline engine/Shopware/Controllers/Backend/Analytics.php:284]: You can wrap the array_splice in an if condition format != csv. I guess it is better to read.

File: engine/Shopware/Controllers/Backend/Analytics.php [hunk 2]
```diff
@@ -279,11 +279,17 @@ public function getRatingAction()
...
+            $limit = \count($data);
+        }
+
         $values = array_values($data);
         $splice = array_splice(
             $values,
-            (int) $this->Request()->getParam('start', 0),
```
Comment (mitelg) [inline engine/Shopware/Controllers/Backend/Analytics.php:285]: why did you remove the cast?

**Comment by mitelg** [comment]
would you mind adding a test for this action here? https://github.com/shopware/shopware/blob/5.7/tests/Functional/Controllers/Backend/AnalyticsTest.php

File: engine/Shopware/Controllers/Backend/Analytics.php (hunk 2 above)
Comment (resslinger) [reply engine/Shopware/Controllers/Backend/Analytics.php:285]: I copied the logic from the method getOverviewAction before. I will add the cast on this method too.

File: engine/Shopware/Controllers/Backend/Analytics.php (hunk 1 above)
Comment (resslinger) [reply engine/Shopware/Controllers/Backend/Analytics.php:284]: I don't think, that a wrap make it easier to read. The array_splice is needed with format == csv and format != csv. So it duplicates the code.

**Comment by resslinger** [comment]
@mitelg I'll try it... :)

**Comment by mitelg** [comment]
let us know, if you need help with it :blush: :+1:

**Comment by philipreinken** [comment]
Hello, we're closing this pull request due to inactivity. If this change is still important to you, feel free to open another pull request.

For more information about our contribution guidelines, see https://developers.shopware.com/community/contribution-guideline

**Comment by resslinger** [comment]
@philipgatzka I only have not enough time to add a test. But the bug exists furthermore.

**Comment by philipreinken** [comment]
Got you, and thank you for investing the time - it's just that we don't want old pull requests to pile up over time and we, therefore, clean them up now and then. Please open another PR (or re-open this one) once you find the time to complete it 🙂
